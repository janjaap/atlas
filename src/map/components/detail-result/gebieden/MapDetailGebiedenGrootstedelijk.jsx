import React from 'react';
import PropTypes from 'prop-types';

import MapDetailResultWrapper from '../MapDetailResultWrapper';

const MapDetailGebiedenGrootstedelijk = ({
  panoUrl, grootstedelijk, onMaximize, onPanoPreviewClick
}) => (
  <MapDetailResultWrapper
    panoUrl={panoUrl}
    onMaximize={onMaximize}
    onPanoPreviewClick={onPanoPreviewClick}
    subTitle={grootstedelijk.label}
    title="Grootstedelijk gebied"
  />
);

MapDetailGebiedenGrootstedelijk.propTypes = {
  grootstedelijk: PropTypes.shape({
    label: PropTypes.string.isRequired
  }).isRequired,
  panoUrl: PropTypes.string.isRequired,
  onMaximize: PropTypes.func.isRequired,
  onPanoPreviewClick: PropTypes.func.isRequired
};

export default MapDetailGebiedenGrootstedelijk;
