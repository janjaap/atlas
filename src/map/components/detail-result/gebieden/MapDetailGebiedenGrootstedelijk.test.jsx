import React from 'react';
import { shallow } from 'enzyme';

import MapDetailGebiedenGrootstedelijk from './MapDetailGebiedenGrootstedelijk';

describe('MapDetailGebiedenGrootstedelijk', () => {
  it('should render everything', () => {
    const grootstedelijk = {
      label: 'Grootstedelijk label'
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailGebiedenGrootstedelijk
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        grootstedelijk={grootstedelijk}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
