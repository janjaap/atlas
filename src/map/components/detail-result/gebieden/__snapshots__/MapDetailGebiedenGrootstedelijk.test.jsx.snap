// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDetailGebiedenGrootstedelijk should render everything 1`] = `
<MapDetailResultWrapper
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Grootstedelijk label"
  title="Grootstedelijk gebied"
/>
`;
